src/back/glsl/mod.rs

```diff
@@ -341,7 +341,10 @@ impl<'a, W: Write> Writer<'a, W> {
     ///
     /// # Panics
     /// Might panic if the module is invalid
-    pub fn write(&mut self) -> Result<FastHashMap<String, TextureMapping>, Error> {
+    pub fn write(
+        &mut self,
+        analysis: &Analysis,
```

Comment:
I think it would be better to store `analysis` in the `Writer` itself since we will use it in others parts (for example when writing functions)